float8_experimental/fsdp_utils.py

```diff
@@ -110,3 +112,181 @@ def fsdp_post_all_gather(
             out._scale = scale
             return
         return Float8Tensor(data, scale, param_dtype, self._mm_config), (data,)
+
+
+class WeightWithDelayedFloat8CastTensor(torch.Tensor):
```

Comment:
[no change needed] I wish there was a way to share some more code with the dynamic linear version

Reply:
yeah, me too.  Looking at the code below, really the only code which would be shared is `fsdp_post_all_gather`, everything else would have to have if/else branches for delayed vs dynamic